File: RepoTransBench/target_projects/Java/C#/rohanpadhye_vasco/public_tests/DataFlowSolutionPublicTest.cs
using System.Collections.Generic;
using Xunit;

namespace Vasco.Tests.Public
{
    public class DataFlowSolutionPublicTest
    {
        [Fact]
        public void TestDifferentInOutValues()
        {
            var inDict = new Dictionary<string, int> { { "A", 111 }, { "B", 222 } };
            var outDict = new Dictionary<string, int> { { "A", 777 }, { "B", 888 } };
            var dfs = new DataFlowSolution<string, int>(inDict, outDict);

            Assert.Equal(111, dfs.GetValueBefore("A"));
            Assert.Equal(222, dfs.GetValueBefore("B"));
            Assert.Equal(777, dfs.GetValueAfter("A"));
            Assert.Equal(888, dfs.GetValueAfter("B"));
        }

        [Fact]
        public void TestNullReturnFromMaps()
        {
            var inDict = new Dictionary<string, int>();
            var outDict = new Dictionary<string, int>();
            var dfs = new DataFlowSolution<string, int>(inDict, outDict);

            Assert.Null(dfs.GetValueBefore("notPresent"));
            Assert.Null(dfs.GetValueAfter("notPresent"));
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/rohanpadhye_vasco/public_tests/dataflowsolution_public_test.go
package public_tests

import (
	"testing"
	"vasco"
)

func TestDifferentInOutValues(t *testing.T) {
	in := map[string]int{"A": 111, "B": 222}
	out := map[string]int{"A": 777, "B": 888}
	dfs := vasco.NewDataFlowSolutionInOut[string, int](in, out)

	if dfs.GetValueBefore("A") != 111 {
		t.Error("value before A")
	}
	if dfs.GetValueBefore("B") != 222 {
		t.Error("value before B")
	}
	if dfs.GetValueAfter("A") != 777 {
		t.Error("after A")
	}
	if dfs.GetValueAfter("B") != 888 {
		t.Error("after B")
	}
}

func TestNullReturnFromMaps(t *testing.T) {
	in := map[string]int{}
	out := map[string]int{}
	dfs := vasco.NewDataFlowSolutionInOut[string, int](in, out)

	if dfs.GetValueBefore("notPresent") != 0 {
		t.Error("Should return zero value for missing")
	}
	if dfs.GetValueAfter("notPresent") != 0 {
		t.Error("Should return zero value for missing")
	}
}